### PR TITLE
Permutive AMP Integration

### PR DESCRIPTION
**Pull request by faresite**
## What does this change?
One of the most interesting changes is referencing a Guardian hosted iframe since we are only allowed one`<amp-iframe />` to use for analytics or tracking, so this was our work around to get both Prebid User Sync and Permutive working at the same time.

This is the [PR](https://github.com/guardian/frontend/pull/22242) for hosting the html file in frontend.

## Supporting Links
- [Trello Card](https://trello.com/c/CEmLV8JA)
- [Permutive AMP docs](https://developer.permutive.com/docs/amp)
- [AMP `<amp-iframe/>` docs](https://amp.dev/documentation/components/amp-iframe/)
- [Prebid User Sync](http://prebid.org/dev-docs/show-prebid-ads-on-amp-pages.html#user-sync)

## Why?
why

### CODE REVIEW

File: src/amp/lib/permutive.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                  .map(s => s.trim())
+                  .join()
+            : null;
+    const keywords =
+        rawConfig.keywords && typeof rawConfig.keywords === 'string'
+            ? rawConfig.keywords
+                  .split(',')
+                  .map(s => s.trim())
+                  .join()
+            : null;
```
Comment (buck06191) [inline src/amp/lib/permutive.ts]: Could we extract out the repeated code here? Something like:
```typescript
const cleanStringField = (stringField: string) =>{
    return stringField && typeof stringField === 'string'
            ? stringField
                  .split(',')
                  .map(s => s.trim())
                  .join()
            : null;
}
```

File: src/amp/components/AnalyticsIframe.tsx [hunk 2]
```diff
@@ -1,9 +1,6 @@
 import React from 'react';
 import { css } from 'emotion';
 
-const prebidSrc =
```
Comment (buck06191) [inline src/amp/components/AnalyticsIframe.tsx]: By removing this, does it remove prebid completely? Or has this been added in again elsewhere?

File: src/amp/lib/permutive.ts (hunk 1 above)
Comment (buck06191) [inline src/amp/lib/permutive.ts]: Could we extract out the repeated code here? Something like:
```typescript
const cleanStringField = (stringField: string): string =>{
    return stringField && typeof stringField === 'string'
            ? stringField
                  .split(',')
                  .map(s => s.trim())
                  .join()
            : null;
}

const authors = cleanStringField(rawConfig.author);
```

File: index.d.ts [hunk 3]
```diff
@@ -346,13 +346,27 @@ type OnwardsType = {
     trails: TrailType[];
 };
 
+interface CommercialConfigType {
+    isPaidContent?: boolean;
+    pageId: string;
+    webPublicationDate?: number;
```
Comment (oliverlloyd) [inline index.d.ts]: Not sure if this matters or not but I've only ever seen `webPublicationDate` as a string. I've heard rumours that we have dates in two different formats in CAPI so maybe this is what we're seeing here? If so, is it possible to force it to a string on the server? Or maybe use a different name?

File: src/amp/lib/permutive.ts (hunk 1 above)
Comment (oliverlloyd) [reply src/amp/lib/permutive.ts]: The `typeof stringField === 'string'` or `typeof rawConfig.keywords === 'string'` aren't needed as we have typescript. Go typescript!

File: src/amp/components/AnalyticsIframe.tsx (hunk 2 above)
Comment (faresite) [reply src/amp/components/AnalyticsIframe.tsx]: We're pulling this from the config as a prop like `<AdUserSync url={config.ampIframeUrl} />`

File: src/amp/lib/permutive.ts (hunk 1 above)
Comment (faresite) [reply src/amp/lib/permutive.ts]: @oliverlloyd we don't have a guarantee though as it comes from the frontend config? Correct me if I'm wrong but I was sure TS doesn't check in runtime?!
@buck06191 Great shout, I tend to extract if it's repeated more than 2-3 times and there's no guarantee that `authors` and `keywords` are treated the same in CAPI.

File: index.d.ts (hunk 3 above)
Comment (faresite) [reply index.d.ts]: I hear you and I've noticed that before too ... the webPublicationDate comes in the page config as a number .. Don't think it's up to me to change stuff as who knows what that could break 🤷‍♂️

**Pull request merged**